pkg/solace/errors.go

```diff
@@ -88,6 +88,12 @@ type IllegalArgumentError struct {
 	solaceError
 }
 
+// InvalidDataTypeError indicates an argument with an invalid data type was passed to a function.
```

Comment:
If we remove the check above for string type, we can remove this error. If we decide to keep it then ignore this comment.